### ISSUE TITLE
Time set incorrectly in Node 12

### ISSUE THREAD

**Issue by knightjdr** [tschaub/mock-fs#276]
If I use Node 10, I don't encounter any issues. But using Node 12 (specifically 12.10.0), the file times aren't being set correctly.

If I create a file like this:
```
mockFS.file({
  content: '',
})
```

fs.stat returns 31 Aug, 51738 for the atime and mtime, and invalid for ctime.
```
Stats {
  dev: 8675309,
  mode: 33206,
  nlink: 1,
  uid: 501,
  gid: 20,
  rdev: 0,
  blksize: 4096,
  ino: 13,
  size: 0,
  blocks: 0,
  atimeMs: 1570547350139547.2,
  mtimeMs: 1570547350139547.2,
  ctimeMs: NaN,
  birthtimeMs: NaN,
  atime: +051738-08-31T08:48:59.547Z,
  mtime: +051738-08-31T08:48:59.547Z,
  ctime: Invalid Date,
  birthtime: Invalid Date
}
```

Explicitly setting the atime, ctime, mtime and birthtime has not effect, and produces a similar result.
```
const now = new Date();
mockFS.file({
  content: '',
  atime: now,
  ctime: now,
  mtime: now,
  birthtime: now,
})
```

**Comment by yuraxdrumz** [comment]
Hey,
After reading in nodejs [changelog](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V12.md#12.10.0), I saw a change in fs.stats. [This](https://github.com/nodejs/node/pull/21387) is the pull request from the nodejs repo. Maybe its related.

**Comment by knightjdr** [comment]
Thanks. I'm testing on OSX, but I'll try tonight on Linux and see if I get a different result.

It looks like that change you highlight is the issue. 12.9.0 works fine, but not 12.10.0 (or the latest 12.11.1).

**Comment by knightjdr** [comment]
Same problem on Ubuntu 18.04.3 LTS, with Node 12.10.0.

**Comment by tschaub** [comment]
Fix published in `mock-fs@4.10.2`.  Thanks @3cp.
